src/ParcelRegistry/Parcel/Parcel.cs

```diff
@@ -114,6 +114,36 @@ public void DetachAddressBecauseAddressWasRetired(AddressPersistentLocalId addre
             ApplyChange(new ParcelAddressWasDetachedBecauseAddressWasRetired(ParcelId, CaPaKey, addressPersistentLocalId));
         }
 
+        public void DetachAddressBecauseStreetNameWasReaddressed(AddressPersistentLocalId addressPersistentLocalId)
+        {
+            if (!AddressPersistentLocalIds.Contains(addressPersistentLocalId))
+            {
+                return;
+            }
+
+            ApplyChange(new ParcelAddressWasDetachedBecauseStreetNameWasReaddressed(ParcelId, CaPaKey, addressPersistentLocalId));
+        }
+
+        public void AttachAddressBecauseStreetNameWasReaddressed(AddressPersistentLocalId addressPersistentLocalId)
+        {
+            if (IsRemoved)
```

Comment:
@ArneD I don't know if it's necessary to perform the IsRemoved and !Realized checks? I do think we don't want to throw in case they are.